Library/Formula/qt5.rb

```diff
@@ -72,6 +74,13 @@ class Qt5 < Formula
   depends_on OracleHomeVarRequirement if build.with? "oci"
 
   def install
+    # Save users some frustration by telling them that `--cc=gcc-5` or similar
+    # won't work. It won't be picked up by the Qt build system, there is no easy
+    # way to select an arbitrary compiler, and it's not supported anyway.
+    if ENV.compiler != :clang
```

Comment:
> I think probably `fails_with :gcc => "5"` with a decent enough message will be enough for now.

When you talk about the message you mean the code comment I would put inside the formula code above the `fails_with` line, aren't you?

Also, some bad news I guess: The restrictions implied by `fails_with` aren't checked if a compiler is manually selected via `--cc`. However, they are checked if the compiler is selected via the `HOMEBREW_CC` environment variable or auto-selected. Is this a bug or by design? (The [relevant code](https://github.com/Homebrew/homebrew/blob/06cb002431c05417454d2fd2283d722144d24b5e/Library/Homebrew/extend/ENV/shared.rb#L151-L163) is in `extend/ENV/shared.rb`. Observe how these cases are treated differently.)

> When you get bored making Qt respect superenv seems like a good wee project.

I'm not sure if this is a feasible project. A lot of compiler details seem to be baked in into those `mkspecs` and I can easily imagine things to break if `superenv` is forced on the Qt build system.